.github/workflows/_rust.yml

```diff
@@ -102,10 +102,12 @@ jobs:
           # First, run all tests.
           cargo test --all-features ${{ steps.setup-rust.outputs.packages }} -- --include-ignored --nocapture
 
-          # Assert that we sample certain transitions
+          # Assert that we hit certain codepaths
+          # Here, we can simply grep the log directory for whatever we want.
           rg SendICMPPacketToCidrResource $TESTCASES_DIR
           rg SendICMPPacketToDnsResource $TESTCASES_DIR
           rg SendDnsQueries $TESTCASES_DIR
+          rg "Still connecting to gateway, buffering allow access" $TESTCASES_DIR
```

Comment:
@conectado It might be a bit unusual but I think simply grepping all test logs to check that we at least hit this case once is a pretty good "80%" solution for now to ensure we execute a certain code path in the tests.

We do have 1 regression seed that hits this so this shouldn't fail randomly in CI.